data_structures/src/chain.rs

```diff
@@ -3004,6 +3005,8 @@ pub struct ChainState {
     pub alt_keys: AltKeys,
     /// Current superblock state
     pub superblock_state: SuperBlockState,
+    /// WIPs votes counter by bits
+    pub bit_wip_counter: HashMap<i8, WipVotesCounter>,
```

Comment:
Use an array instead?

```suggestion
    pub bit_wip_counter: [WipVotesCounter; 32],
```